esmvalcore/preprocessor/_regrid.py

```diff
@@ -1086,3 +1086,40 @@ def get_reference_levels(filename, project, dataset, short_name, mip,
     except iris.exceptions.CoordinateNotFoundError:
         raise ValueError('z-coord not available in {}'.format(filename))
     return coord.points.tolist()
+
+
+def extract(cube, definition, scheme):
+    """Extract a point in any coordinates with a given interpolation scheme.
+
+    Multiple points can also be extracted, by supplying an array of
+    coordinates. The resulting point cube will match the respective
+    coordinates to those of the input coordinates.
+    If the input coordinate is a scalar, the dimension will be a
+    scalar in the output cube.
+
+    Parameters
+    ----------
+    cube : cube
+        The source cube to extract a point from.
+    defintion : dict(str, float or array of float)
+        The coordinate - values pairs to extract
+    scheme : str
+        The interpolation scheme. 'linear' or 'nearest'. No default.
+
+    Returns
+    -------
+    Returns a cube with the extracted point(s), and with adjusted
+    coordinates (see above).
+
+    Raises
+    ------
+    ValueError:
+        If the interpolation scheme is not provided or is not recognised.
+    """
```

Comment:
wait - what if I am trying to extract latitude: 80 from a cube that's on (-40, 40) lat only?

Reply:
sorry, should have mentioned - in this case I'm missing half the data so the extrapolation will be rather inaccurate

Reply:
(just took a look) - it's bonkers, there are no restrictions on the location of the point(s) wrt to the actual given data and I *think* [`cube.interpolate()`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/cube.html?highlight=interpolate#iris.cube.Cube.interpolate) will not complain, and rather produce some very synthetic extrapolated data - in any case, this should not concern this PR since we already have the other extractors, am just being anal about something I should have been long time ago :grin: